src/shared/bus-polkit.c

```diff
@@ -284,15 +296,19 @@ static int process_polkit_response(
         assert(action);
         assert(ret_error);
 
-        assert(q->action);
-        assert(q->reply);
+        /* If we're checking a different action, it must either be one that's already been authorized
+         * or one that we haven't queried yet. */
+        if (!streq_ptr(q->action, action))
+                return strv_contains(q->authorized_actions, action);
 
         /* If the operation we want to authenticate changed between the first and the second time,
          * let's not use this authentication, it might be out of date as the object and context we
          * operate on might have changed. */
-        if (!streq(q->action, action) || !strv_equal(q->details, (char**) details))
+        if (!strv_equal(q->details, (char**) details))
                 return -ESTALE;
```

Comment:
so this ESTALE logic should go away entirely. instead: if we don't find the authorization we are looking for in the list yet, we ask polkit for it, and then add it to the list. Hence if the previous request doesn't match the new request the strategy should now be: "ok, then let's ask that too" rather than "uh, fuck you, take an ESTALE!"